File: src/main/java/io/github/bot/terminal/application/domains/common/conditions/entity/Condition.java
package io.github.bot.terminal.application.domains.common.conditions.entity;

import io.github.bot.terminal.application.domains.common.conditions.values.ConditionType;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class Condition {
    private ConditionType conditionType;
    private Double expectedValue;

    public boolean check(Double value) {
        return conditionType.check(value, expectedValue);
    }
}

File: src/main/kotlin/io/github/bot/terminal/application/domains/common/conditions/entity/Condition.kt
package io.github.bot.terminal.application.domains.common.conditions.entity

import io.github.bot.terminal.application.domains.common.conditions.values.ConditionType

class Condition(
        val conditionType: ConditionType,
        val expectedValue: Double
) {

    fun check(value: Double): Boolean {
        return conditionType.check(value, expectedValue)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Condition

        if (conditionType != other.conditionType) return false
        if (expectedValue != other.expectedValue) return false

        return true
    }

    override fun hashCode(): Int {
        var result = conditionType.hashCode()
        result = 31 * result + expectedValue.hashCode()
        return result
    }
}